### ISSUE TITLE
Port forwarding not working over remote tunnel

### ISSUE THREAD

**Issue by msdkhairi** [microsoft/vscode-remote-release#10370]
When I try to forward a port, I receive an error stating, "The request was denied because it would exceed the limit for 'PortsPerTunnel' (10)." However, I have not forwarded any ports, as shown in the image below:

![image](https://github.com/user-attachments/assets/1284b88f-fbe4-4ec8-8e07-c0718317c117)

- VSCode Version: 1.94.1 (Darwin arm64 24.0.0)
- Local OS Version: macOS 15.0.1
- Remote OS Version: Docker 20.10.15 on an Ubuntu 20.04.4 LTS
- Remote Extension/Connection Type: Tunnel

- Logs: 
Unable to forward localhost:6006. Could not forward port: Error: WrappedError(WrappedError { message: "error adding port to relay", original: "failed to add port to tunnel: response error: HTTP status 429 Too Many Requests from https://usw2.rel.tunnels.api.visualstudio.com/tunnels/[...]/ports/6006?api-version=2023-09-27-preview (request ID [...]-[...]-[...]-[...]-[...]): {\"title\":\"Resource limit exceeded.\",\"status\":429,\"detail\":\"The request was denied because it would exceed the limit for 'PortsPerTunnel' (10).\",\"traceId\":\"00-[...]-[...]-00\"}" })

Steps to Reproduce:

1. Set up a VSCode tunnel in Docker on a remote server.
2. Access the tunnel on a local machine and attempt to forward a port.

**Comment by DavidAGInnovation** [comment]
Hi, I'm having the same error. Let's hope we have a solution soon.

**Comment by einmalick** [comment]
I'm encountering the same issue. 

The project where I first noticed it has lot of initial requests upon access, and it seems to be related to a rate limit. To test further, I created a small Create React App. Initially, the first few page reloads work fine, but then, I start getting "429 Too Many Requests" errors.

**Comment by pjninnim** [comment]
I'm encountering the same issue as described above.

**Comment by einmalick** [comment]
I believe the error is likely related to rate limits. After a few test runs, it seems that I am either exceeding or nearing the tunnel limits when checking the dev tools. I will investigate further."

Meanwhile you could also check if you are reaching the limits via your DEV-Tools network inspect.

Here are the Dev-Tool-Limits provided by MS: [Link](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#dev-tunnels-limits) 

![Image](https://github.com/user-attachments/assets/4395a8ea-e42b-4744-adf9-0b37efc4c549)

**Comment by pjninnim** [comment]
I don't think the limits can be getting reset:
![Image](https://github.com/user-attachments/assets/59567b1c-1b02-4734-a187-f37a87c7a79b)

**Comment by Jahtila** [comment]
I'm encountering the same issue as well. Initially it worked for me for a while, but now experiencing the same 429 status code.

**Comment by einmalick** [comment]
It could also be possible that the bandwidth limit was reached. It is 5gb per month.

**Comment by aleleba** [comment]
It happened to me too.

**Comment by connor4312** [comment]
Hi, this is being discussed on https://github.com/microsoft/dev-tunnels/issues/491. The limit you were hitting should now be raised.

**Comment by msdkhairi** [comment]
Thank you, @connor4312. Unfortunately, the update didn't resolve the issue in my case. A workaround I found is to stop the tunnel and start a new one. From what I understand, it seems that some old port forwards remain active in the background, even though they are not shown in the vscode ui, which is likely why the error "exceed the limit for 'PortsPerTunnel'" is raised.

**Comment by connor4312** [comment]
Yes, the `PortsPerTunnel` is a different issue than the one the comments were referring to. THat's probably a bug on my end.

**Comment by connor4312** [comment]
@msdkhairi can you please share the "Port Forward" output from the Output view in the bottom panel the first time you try to forward a port? We have logic that cleans up any existing ports when starting a new forwarding session, but maybe that's failing for some reason.

**Comment by msdkhairi** [comment]
I don't see a Port Forwarding window in the output window options. This is the menu I have:

`Git, GitHub, GitHub Authentication, GitHub Copilot, GitHub Copilot Chat, GitHub Copilot Log, GitHub Pull Request, Microsoft Authentication, Python, Python Language Server, Python Locator, Python Test Log, Remote - Tunnels, Cloud Changes, Extension Host, Extension Host (Remote), Main, Notebook, Pty Host, Remote Tunnel Service, Server, Settings Sync, Shared, Tasks, Terminal, Window`

**Comment by connor4312** [comment]
Oh, my apologies, I thought you were using the built-in port forwarding, missed that this was over remote - tunnels.

**Comment by veryCrunchy** [comment]
# Solution/Workaround
Uninstall the tunnel service, this will remove all lingering ports.
<details>
<summary>Details</summary>

These ports are not visible, since they are no longer open locally.
They could be lingering due to unexpected disconnects, as a result the api is not called.
The lingering ports will stack up until you reach the limit.

</details>

```
code tunnel service uninstall
```
Then simply reinstall the service
```
code tunnel service install
```
You'll be able to reconnect and pick up where you left from, now with port forwarding working again!

**Comment by DavidAGInnovation** [comment]
Thanks for the solution! It worked @veryCrunchy

**Comment by DavidAGInnovation** [comment]
UPDATE: Initially, it solved the issue for me, but the problem reappeared, and that fix didn't work anymore. I've found a definitive solution for me.

Click on the Install Server VS Code button from this website: https://www.npmjs.com/package/@playwright/mcp

<img width="802" height="411" alt="Image" src="https://github.com/user-attachments/assets/f05e8c4f-dafb-4790-963c-acdc69f53bd8" />

That should fix it!

**Comment by katoue** [comment]
Reinstalling the tunnel service did not work for me. I was able to resolve the issue by restarting the remote VS Code tunnel service, which clears any lingering or “ghost” ports that may be causing the PortsPerTunnel limit error:
`pkill -f "code tunnel"`
`code tunnel`
